### ISSUE TITLE
Recent change breaks branch analysis reporting on SonarQube EE

### ISSUE THREAD

**Issue by Hydragyrum** [gabrie-allaigre/sonar-gitlab-plugin#232]
A change was recently added with #227 where the existence of the "branch" plugin is checked.
(SonarFacade.java : 273)
```java
 if (isNotBlankAndNotEmpty(branch) && isPluginInstalled("branch"))
```
It seems that on the Developer editions and higher, this plugin is not shown in the list of installed plugins (as it's not a plugin on these versions). As a result, the code never sets the proper branch when searching for issues, and thus all issues are brought down to Gitlab CI, failing the pipeline.

eg:

![image](https://user-images.githubusercontent.com/4928181/55803723-31b82480-5adb-11e9-8710-2dae5ba3bc51.png)

The quality gate passes, but because there are issues (unrelated to the branch in question), the pipeline fails.

**Comment by Hydragyrum** [comment]
some additional info, apparently this was changed in v7.2
https://docs.sonarqube.org/display/PLUG/Plugin+Version+Matrix

**Comment by shweta-chadha** [comment]
@gabrie-allaigre - Even we are facing a similar issue in our project. We are using **SonarQube developer edition - version 7.7** and **gitlab plugin version 4.1.0**.

**Comment by asircar** [comment]
I have created a merge request for the same, please review and merge. https://github.com/gabrie-allaigre/sonar-gitlab-plugin/pull/236

**Comment by Hydragyrum** [comment]
The problem with that PR is that it may adversely affect community users, who don't have the branch plugin.

**Comment by asircar** [comment]
I agree that this may adversely impact the community users but I feel that we should think of fixing the issue with the community edition in a cleaner way without checking for the branch plugin as that fix is breaking the developer and higher editions. That's the reason I think that we should go back to the previous state and then address the initial problem in a better way. We are using the developer edition and are very badly impacted by this issue.

**Comment by Hydragyrum** [comment]
I agree as we're also using the enterprise edition. The workaround for the community edition is to not use the branch.name option to begin with. 

Ultimately, it's up to @gabrie-allaigre to decide.

**Comment by Hydragyrum** [comment]
The ideal solution would be to bake in support for Gitlab into Sonarqube like they did for Github

**Comment by LennyPenny** [comment]
@Hydragyrum is there any tracking issue for that? 
Just mentioned it to them here with regard to sonarcloud https://community.sonarsource.com/t/merge-requests-support-for-gitlab/806/2

**Comment by Hydragyrum** [comment]
@LennyPenny I linked a gitlab issue in that discussion as well. For now I think we have to push Gitlab to work with Sonar on this.

**Comment by Hydragyrum** [comment]
https://gitlab.com/gitlab-org/gitlab-ce/issues/25238 for reference

**Comment by asircar** [comment]
@gabrie-allaigre Can you please merge this?

**Comment by gabrie-allaigre** [comment]
Ok, thanks
